### PR TITLE
Better handing when the pucks lose tracking and openvr still send you a "valid" data

### PR DESCRIPTION
**Pull request by druiz17**
Better handling when the pucks lose tracking and openvr still send you a "valid" data. we ignore the data and send the previously tracked pose until openvr send us the correct data.

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan. 
- having pucks on and set to any configuration
- block the pucks from the sight of the trackers
- you should see the puck freeze in position for a few seconds, then the joint return to the default position

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3f5a13d5a9cbb386dbeca8abee1b620f9ac4b875 available at: https://gist.github.com/bb6404552b9eb888e0d814a69eda8116<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10451-3f5a13d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10451-3f5a13d.dmg

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 1]
```diff
@@ -69,6 +66,28 @@ static bool sortPucksYPosition(std::pair<uint32_t, controller::Pose> firstPuck,
     return (firstPuck.second.translation.y < secondPuck.second.translation.y);
 }
 
+static QString deviceTrackingResultToString(vr::ETrackingResult trackingResult) {
+    QString result;
+    switch (trackingResult) {
```
Comment (ZappoMan) [inline plugins/openvr/src/ViveControllerManager.cpp]: seems like this could be done with a table lookup or array?

Comment (druiz17) [inline plugins/openvr/src/ViveControllerManager.cpp]: I think that might be a better approach.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e6020e0137ae2e9abd0d1b88c235efc62002b498 available at: https://gist.github.com/bb6404552b9eb888e0d814a69eda8116<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10451-e6020e0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10451-e6020e0.dmg

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 2]
```diff
@@ -69,6 +74,16 @@ static bool sortPucksYPosition(std::pair<uint32_t, controller::Pose> firstPuck,
     return (firstPuck.second.translation.y < secondPuck.second.translation.y);
 }
 
+static QString deviceTrackingResultToString(vr::ETrackingResult trackingResult) {
+    QString result;
```
Comment (ZappoMan) [inline plugins/openvr/src/ViveControllerManager.cpp]: maybe this should be "unknown"?

**Comment by anshuman64** [comment]
I think this is pucking fantastic! Did not get into an crazy/impossible position even once due to lost tracking. Only noticeable effect is the "lag" of body parts, but I guess that is inevitable. I believe this is a game changer in our performances.

Couple videos of me in action:
Front View: https://drive.google.com/open?id=0B_nmInIG19-uRnRCd2pCYUFqU2M
Back View: https://drive.google.com/open?id=0B_nmInIG19-uRmlhNWhaMF9tRm8

**Pull request merged**